systemd/qubes-qrexec-agent.service

```diff
@@ -5,6 +5,7 @@ After=xendriverdomain.service systemd-user-sessions.service
 [Service]
 Type=notify
 ExecStartPre=+/bin/sh -c '[ -e /dev/xen/evtchn ] || modprobe xen_evtchn'
+ExecStartPre=-+/bin/sh -c '[ ! -d /run/qubes/rpc-config ] && mkdir -Z etc_t /run/qubes/rpc-config'
```

Comment:
Is that -Z really needed? Or is that mkdir even appropriate here? Generally, SELinux should set correct labels when you create the directory yourself (with the above change in .fc file). And if pre-creating the directory is necessary anyway, better do that via tmpfiles mechanism.

Reply:
At least when I try making the directory manually in a shell, it ends up with the type `qubes_var_run_t`, despite the change. I can run `sudo restorecon /run/qubes/rpc-config` to fix it, but that's hacky.

Will try the systemd-tmpfiles mechanism to see if it works better.

Reply:
Implemented, tested, works.